hpy/devel/include/hpy/cpython/misc.h

```diff
@@ -194,6 +203,27 @@ HPyAPI_FUNC void HPy_Close(HPyContext *ctx, HPy handle)
     Py_XDECREF(_h2py(handle));
 }
 
+HPyAPI_FUNC HPyField HPyField_Store(HPyContext *ctx, HPy h)
```

Comment:
I am wondering what happens with code like:
```
type_struct.x = HPyField_Store(ctx, h);
```
Is it possible that there is briefly a temporary variable somewhere containing an HPyField that the GC doesn't know about? Or the temptation for the user to write `if (....) { a = HPyField_Store(ctx, h); } else { a = HPyField_Store(ctx, h) } type_struct.x = a`?

It feels safer if `HPyField_Store` takes the address to store the address in directly. Then the implementation can do whatever is necessary to ensure the pointer is definitely stored safely.

Reply:
I think this is a comment about old code? The new signature for `HPyField_Store` does indeed what you suggest, see `public_api.h`:
```c
void HPyField_Store(HPyContext *ctx, HPyField *target_field, HPy h);
```

Reply:
ouch, my fault. I had local commits which weren't pushed :(
Anyway, if you look at what I just pushed, you see I have already addressed this :)